pysilcam/__main__.py

```diff
@@ -156,45 +147,66 @@ def silcam_acquire(datapath, config_filename, writeToDisk=True, gui=None):
-            gui.put_nowait((timestamp, imraw, imraw, rtdict))
+    acq = Acquire(USE_PYMBA=True, datapath=datapath, writeToDisk=writeToDisk, gui=gui)  # ini class
+
+    acq.stream_images(camera_config_file=config_filename)
+
+
+def silcam_acquire(datapath, config_filename, writeToDisk=True, gui=None):
+    '''dev for processing with VimbaPython
+
+    Args:
+       datapath              (str)          :  Path to the image storage
+       config_filename=None  (str)          :  Camera config file
+       writeToDisk=True      (Bool)         :  True will enable writing of raw data to disc
+                                               False will disable writing of raw data to disc
+       gui=None          (Class object)     :  Queue used to pass information between process thread and GUI
+                                               initialised in ProcThread within guicals.py
+    '''
+
+    # Load the configuration, create settings object
+    settings = PySilcamSettings(config_filename)
+
+    # Print configuration to screen
+    print('---- CONFIGURATION ----\n')
+    settings.config.write(sys.stdout)
+    print('-----------------------\n')
+
+    if writeToDisk:
+        # Copy config file
+        configFile2Copy = datetime.datetime.now().strftime('D%Y%m%dT%H%M%S.%f') + os.path.basename(config_filename)
+        copyfile(config_filename, os.path.join(datapath, configFile2Copy))
+
+    configure_logger(settings.General)
+    logger = logging.getLogger(__name__ + '.silcam_acquire')
+
+    # update path_length
+    updatePathLength(settings, logger)
+
+    # create a last in first out queue eventually to be part of the background module
+    # manager = MyManager()
+    # manager.start()
+    # raw_image_queue = manager.LifoQueue(2)  # limit queue size to two images
+    raw_image_queue = multiprocessing.Queue(2)
+    logger.info("raw image queue initialised.")
+
+    backgrounder = Backgrounder(settings.Background.num_images,
+                                bad_lighting_limit=settings.Process.bad_lighting_limit,
+                                real_time_stats=settings.Process.real_time_stats)
+    backgrounder_process = multiprocessing.Process(target=backgrounder.run, args=(config_filename, raw_image_queue, None))
+    backgrounder_process.start()
+    logger.info("backgrounder_process started.")
+
+    acq = Acquire(USE_PYMBA=False, datapath=datapath, writeToDisk=False, gui=gui, raw_image_queue=raw_image_queue)  # ini class
+
+    acq.stream_images(config_file=config_filename)
+
+    backgrounder_process.join()  # shut down subprocesses after stopping (needs checking how to do this)
 
 
 # the standard processing method under active development
-def silcam_process(config_filename, datapath, multiProcess=True, realtime=False, discWrite=False, nbImages=None,
-                   gui=None,
-                   overwriteSTATS=True):
+def silcam_process(config_filename, datapath, multiProcess=True, realtime=False,
```

Comment:
C901: 'silcam_process' is too complex (12)  <!-- Automatically posted by Lintly -->